app/modules/asset_groups/models.py

```diff
@@ -210,7 +209,25 @@ def commit(self):
                 f'AssetGroupSighting {self.guid} has no metadata',
                 obj=self,
             )
-        cleanup = Cleanup('AssetGroup')
+
+        if 'time' not in self.sighting_config:
```

Comment:
changed order to validate time before crreating sighting and use if valid. Simplified the error handling significantly